v2/pink-sb/src/lib/Root.svelte

```diff
@@ -1,9 +1,14 @@
 <script lang="ts">
     export let theme: Record<string, string>;
 
-    $: Object.keys(theme).forEach((key) => {
-        document.documentElement.style.setProperty(`--${key}`, theme[key]);
-    });
+    $: {
+        const cssVariables = Object.keys(theme)
+            .map((key) => `--${key}: ${theme[key]};`)
+            .join(' ');
+        const styleElement = document.createElement('style');
+        document.head.prepend(styleElement);
+        styleElement.textContent = `:root {${cssVariables}}`;
+    }
```

Comment:
I think this will be creating a style element each time this is triggered.

Maybe we can try and use the `<svelte:head />` special element.

```html
<svelte:head>
	<style>{style}</style>
</svelte:head>
```